stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -0,0 +1,634 @@
+        :UUID => Dict{Base.UUID,Dict{String,Union{String,SHA1}}}(),
+
+        # Overrides by hash
+        :hash => Dict{SHA1,Union{String,SHA1}}(),
+    )
+
+    for override_file in reverse(artifacts_dirs("Overrides.toml"))
+        !isfile(override_file) && continue
+
+        # Load the toml file
+        depot_override_dict = parse_toml(override_file)
+
+        function parse_mapping(mapping::String, name::String)
+            if !isabspath(mapping) && !isempty(mapping)
+                try
+                    mapping = Base.SHA1(mapping)
+                catch e
+                    @error("Invalid override in '$(override_file)': entry '$(name)' must map to an absolute path or SHA1 hash!")
+                    rethrow()
+                end
+            end
+            return mapping
+        end
+        function parse_mapping(mapping::Dict, name::String)
+            return Dict(k => parse_mapping(v, name) for (k, v) in mapping)
+        end
+
+        for (k, mapping) in depot_override_dict
+            # First, parse the mapping. Is it an absolute path, a valid SHA1-hash, or neither?
+            try
+                mapping = parse_mapping(mapping, k)
+            catch
+                @error("Invalid override in '$(override_file)': failed to parse entry `$(k)`")
+                continue
+            end
+
+            # Next, determine if this is a hash override or a UUID/name override
+            if isa(mapping, String) || isa(mapping, SHA1)
+                # if this mapping is a direct mapping (e.g. a String), store it as a hash override
+                hash = try
+                    Base.SHA1(hex2bytes(k))
+                catch
+                    @error("Invalid override in '$(override_file)': Invalid SHA1 hash '$(k)'")
+                    continue
+                end
+
+                # If this mapping is the empty string, un-override it
+                if mapping == ""
+                    delete!(overrides[:hash], hash)
+                else
+                    overrides[:hash][hash] = mapping
+                end
+            elseif isa(mapping, Dict)
+                # Convert `k` into a uuid
+                uuid = try
+                    Base.UUID(k)
+                catch
+                    @error("Invalid override in '$(override_file)': Invalid UUID '$(k)'")
+                    continue
+                end
+
+                # If this mapping is itself a dict, store it as a set of UUID/artifact name overrides
+                ovruuid = overrides[:UUID]::Dict{Base.UUID,Dict{String,Union{String,SHA1}}}
+                if !haskey(ovruuid, uuid)
+                    ovruuid[uuid] = Dict{String,Union{String,SHA1}}()
+                end
+
+                # For each name in the mapping, update appropriately
+                for name in keys(mapping)
+                    # If the mapping for this name is the empty string, un-override it
+                    if mapping[name] == ""
```

Comment:
Why loop over `keys(mapping)` and then immediately lookup `mapping[name]`? Maybe just loop directly over the dict?